quickwit-search/src/root.rs

```diff
@@ -136,39 +142,60 @@ fn analyze_errors(search_result: &SearchResultsByAddr) -> Option<ErrorRetries> {
         .filter(|(_addr, result)| {
             result
                 .as_ref()
-                .map(|ok_res| ok_res.failed_requests.len() as u64 != ok_res.num_attempted_splits)
+                .map(|ok_res| ok_res.failed_splits.len() as u64 != ok_res.num_attempted_splits)
                 .unwrap_or(false)
         })
         .map(|(addr, _)| *addr)
         .collect();
+    debug!("{:?}", &complete_failure_nodes);
+    debug!("{:?}", &partial_or_no_failure_nodes);
 
     // Here we collect the failed requests on the node. It does not yet include failed requests
     // against that node
-    let failed_split_ids = search_result
+    let mut retry_split_ids = search_result
         .values()
         .filter_map(|result| result.as_ref().ok())
         .flat_map(|res| {
             let results = res
-                .failed_requests
+                .failed_splits
                 .iter()
-                .map(move |failed_req| failed_req.split_id.to_string())
+                .filter(move |failed_splits| failed_splits.retryable_error) // only retry splits marked as retryable
+                .map(move |failed_splits| failed_splits.split_id.to_string())
                 .collect_vec();
```

Comment:
I suspect line 164 and 165 are useless?

Reply:
ah yeah good catch, it was necessary in a previous version